sites/partners/src/listings/PaperListingDetails/sections/DetailUnits.tsx

```diff
@@ -1,47 +1,118 @@
 import React, { useContext, useMemo } from "react"
-import { t, GridSection, MinimalTable, Button, ViewItem } from "@bloom-housing/ui-components"
+import { t, GridSection, MinimalTable } from "@bloom-housing/ui-components"
 import { ListingContext } from "../../ListingContext"
 import { UnitDrawer } from "../DetailsUnitDrawer"
+import { MonthlyRentDeterminationType } from "@bloom-housing/backend-core/types"
 
 type DetailUnitsProps = {
   setUnitDrawer: (unit: UnitDrawer) => void
 }
 
+function isDefined(item: number | string) {
+  return item !== null && item !== undefined && item !== ""
+}
+
+function formatRange(min: string | number, max: string | number, prefix: string, postfix: string) {
+  if (!isDefined(min) && !isDefined(max)) return ""
+  if (min == max || !isDefined(max)) return `${prefix}${min}${postfix}`
+  if (!isDefined(min)) return `${prefix}${max}${postfix}`
+  return `${prefix}${min}${postfix} - ${prefix}${max}${postfix}`
```

Comment:
Preference: I'm not a fan of this style.

Also this, and `isDefined` could be used elsewhere if we move it into `shared-helpers`.